dandiapi/api/tests/test_tasks.py

```diff
@@ -105,8 +108,120 @@ def test_write_assets_yaml_already_exists(storage: Storage, version: Version, as
     storage.save(assets_yaml_path, ContentFile(b'wrong contents'))
 
     tasks.write_yamls(version.id)
+    expected = YAMLRenderer().render(
+        [asset.generate_metadata(version) for asset in version.assets.all()]
+    )
 
     with storage.open(assets_yaml_path) as f:
-        assert f.read() == YAMLRenderer().render(
-            [asset.metadata.metadata for asset in version.assets.all()]
-        )
+        assert f.read() == expected
+
+
+@pytest.mark.django_db
+def test_validate_asset_metadata(version: Version, asset: Asset):
+    version.assets.add(asset)
+
+    # This is the minimum asset metadata for schema version 0.3.0
+    asset.metadata.metadata = {
+        'schemaVersion': '0.3.0',
+        'encodingFormat': 'application/x-nwb',
+    }
+    asset.metadata.save()
+
+    tasks.validate_asset_metadata(version.id, asset.id)
+
+    asset.refresh_from_db()
+
+    assert asset.status == Asset.Status.VALID
+    assert asset.validation_error == ''
+
+
+@pytest.mark.django_db
+def test_validate_asset_metadata_no_schema_version(version: Version, asset: Asset):
+    version.assets.add(asset)
+
+    asset.metadata.metadata = {}
+    asset.metadata.save()
+
+    tasks.validate_asset_metadata(version.id, asset.id)
+
+    asset.refresh_from_db()
+
+    assert asset.status == Asset.Status.INVALID
+    assert asset.validation_error == 'schemaVersion not specified'
+
+
+@pytest.mark.django_db
+def test_validate_asset_metadata_malformed_schema_version(version: Version, asset: Asset):
+    version.assets.add(asset)
+
+    asset.metadata.metadata = {
+        'schemaVersion': 'xxx',
+    }
+    asset.metadata.save()
+
+    tasks.validate_asset_metadata(version.id, asset.id)
+
+    asset.refresh_from_db()
+
+    assert asset.status == Asset.Status.INVALID
+    assert asset.validation_error == (
+        '404 Client Error: Not Found for url: '
+        'https://raw.githubusercontent.com/dandi/schema/master/releases/xxx/asset.json'
```

Comment:
I wonder if we should catch this and report it as an invalid schema version?

Reply:
We could, but when relying on an external API like this we still need to handle any possible API errors anyway, so I don't see much point in a special error message. `schemaVersion` will always be set automatically by the meditor or whatever generated the metadata, users shouldn't be setting it themselves anyway.